File: LeetCode/HeightChecker/3228.cs
public class Solution {
    public int HeightChecker(int[] heights) {
        int[] expectedHeights = new int[heights.Length];
        Array.Copy(heights, expectedHeights, heights.Length);
        Array.Sort(expectedHeights);
        int count = 0;
        
        for(int i = 0; i<heights.Length; i++){
            if(heights[i] != expectedHeights[i]){
                count++;
            }
        }
        
        return count;
    }
}

File: LeetCode/HeightChecker/3228.ts
function heightChecker(heights: number[]): number {
  let expectedHeights = [...heights].sort((a, b) => a - b);
  let unmatched = 0;
  for (let i = 0; i < heights.length; i++) {
    if (heights[i] !== expectedHeights[i]) {
      unmatched++;
    }
  }

  return unmatched;
}
